assets/js/Containers/FrevaGPT/index.js

```diff
@@ -0,0 +1,438 @@
+
+    this.state = {
+      loading: false,
+      userInput: "",
+      botModelList: [],
+      botModel: "",
+      hideBotModelList: true,
+      botOkay: undefined,
+      showSuggestions: true,
+      dynamicAnswer: "",
+      dynamicVariant: "",
+    };
+  }
+
+  async componentDidMount() {
+    // if thread giving on mounting the component, set thread within store
+    const givenQueryParams = browserHistory.getCurrentLocation().query;
+    if (
+      has(givenQueryParams, "thread_id") &&
+      !isEmpty(givenQueryParams.thread_id)
+    ) {
+      this.props.dispatch(setThread(givenQueryParams.thread_id));
+
+      // request content of old thread if threa_id is given
+      this.setState({ loading: true });
+      await this.getOldThread(givenQueryParams.thread_id);
+      this.setState({ loading: false, showSuggestions: false });
+    }
+
+    const successfulPing = async () => {
+      let pingSuccessful = false;
+
+      try {
+        const response = await fetch("/api/chatbot/ping");
+        if (response.status === 200) pingSuccessful = true;
+      } catch (err) {
+        console.error("PingError: ", err);
+      }
+
+      return pingSuccessful;
+    };
+
+    const getBotModels = async () => {
+      const queryObject = {
+        auth_key: process.env.BOT_AUTH_KEY,
+      };
+      const response = await fetch(
+        `/api/chatbot/availablechatbots?` + objectToQueryString(queryObject)
+      );
+      this.setState({ botModelList: await response.json() });
+    };
+
+    if (await successfulPing()) {
+      this.setState({ botOkay: true });
+      await getBotModels();
+    } else this.setState({ botOkay: false });
+  }
+
+  createNewChat() {
+    this.props.dispatch(setConversation([]));
+    this.props.dispatch(setThread(""));
+    browserHistory.push({
+      pathname: "/chatbot/",
+      search: "",
+    });
+    this.setState({ showSuggestions: true });
+    window.scrollTo(0, 0);
+  }
+
+  handleUserInput(e) {
+    this.setState({ userInput: e.target.value });
+  }
+
+  async handleKeyDown(e) {
+    if (e.key === "Enter") {
+      this.handleSubmit(e.target.value);
+    }
+  }
+
+  async submitUserInput() {
+    const userInput = this.state.userInput;
+    await this.handleSubmit(userInput);
+    this.setState({ userInput: "" });
+  }
+
+  async handleSubmit(input) {
+    this.props.dispatch(addElement({ variant: "User", content: input }));
+    this.setState({ showSuggestions: false, userInput: "", loading: true });
+
+    try {
+      await this.fetchData(input);
+    } catch (err) {
+      this.props.dispatch(
+        addElement({
+          variant: "FrontendError",
+          content: "An error occured during rendering!",
+        })
+      );
+      // eslint-disable-next-line no-console
+      console.error(err);
+    }
+    this.setState({ loading: false });
+  }
+
+  async fetchData(input) {
+    const queryObject = {
+      input,
+      auth_key: process.env.BOT_AUTH_KEY,
```

Comment:
Having `process.env.BOT_AUTH_KEY` in the react-code means that we are exposing our API-Key to the whole world, which we really don't want. This must be moved to `proxyview.py` instead. The same is true for `freva_config`.